app/logic/Mail/ActiveSync/ActiveSyncFolder.ts

```diff
@@ -163,14 +161,18 @@ export class ActiveSyncFolder extends Folder implements ActiveSyncPingable {
           await this.storage.saveMessage(email);
           newMsgs.add(email);
         }
+      } catch (ex) {
+        this.account.errorCallback(ex);
       }
-      for (let item of ensureArray(response.Commands?.Delete)) {
+      for (let item of ensureArray(response.Commands?.Delete)) try {
```

Comment:
`for` and `try` should be on separate lines and should each have their own `{}` blocks.